lms/djangoapps/instructor_task/api.py

```diff
@@ -437,17 +437,22 @@ def submit_export_ora2_data(request, course_key):
     return submit_task(request, task_type, task_class, course_key, task_input, task_key)
 
 
-def generate_certificates_for_students(request, course_key, students=None):  # pylint: disable=invalid-name
+def generate_certificates_for_students(request, course_key, student_set=None, specific_student_id=None):  # pylint: disable=invalid-name
     """
-    Submits a task to generate certificates for given students enrolled in the course or
-    all students if argument 'students' is None
+    Submits a task to generate certificates for given students enrolled in the course.
 
     Raises AlreadyRunningError if certificates are currently being generated.
+    Raises ValueError if student_set is 'specific_student' and specific_student_id is 'None'
     """
-    if students:
-        task_type = 'generate_certificates_certain_student'
-        students = [student.id for student in students]
-        task_input = {'students': students}
+    if student_set:
+        task_type = 'generate_certificates_student_set'
+        task_input = {'student_set': student_set}
+
+        if student_set == 'specific_student':
+            task_type = 'generate_certificates_certain_student'
+            if not specific_student_id:
+                raise ValueError("Student ID is missing")
```

Comment:
Also make the message more specific — "attempted to generate certificate for a single student, but no specific student id provided"